.github/workflows/cre-local-env-tests.yaml

```diff
@@ -42,9 +45,28 @@ jobs:
           ref: ${{ github.event_name == 'pull_request' && github.sha || inputs.chainlink_version }}
 
       - name: Set up Go
+        id: setup-go
         uses: actions/setup-go@v5
         with:
           go-version-file: core/scripts/go.mod
+          cache: true
+
+      - name: Download Go dependencies
+        id: go-deps
+        run: |
+          cd core/scripts/cre/environment
+          if [ "${{ steps.setup-go.outputs.cache-hit }}" == "true" ]; then
```

Comment:
To avoid bash injection and to ensure that everything in the `run: |` block is valid bash, it's generally better to set env vars for these outputs (or inputs) like:

```
        env:
          SETUP_GO_CACHE_HIT: ${{ steps.setup-go.outputs.cache-hit }}
        run: |
          cd core/scripts/cre/environment
          if [[ "${SETUP_GO_CACHE_HIT}" == "true" ]]; then
```

Also try to prefer `[[` (double brackets) with bash as it's more reliable.